src/Renci.SshNet/SftpClient.cs

```diff
@@ -985,7 +985,7 @@ public async Task UploadFileAsync(Stream input, string path, bool canOverride, C
             CheckDisposedOrNotConnected();
             cancellationToken.ThrowIfCancellationRequested();
 
-            using (SftpFileStream output = await SftpFileStream.OpenAsync(_sftpSession, path, canOverride ? FileMode.Create : FileMode.CreateNew, FileAccess.Write, (int)_bufferSize, cancellationToken).ConfigureAwait(false))
+            using (SftpFileStream output = await SftpFileStream.OpenAsync(_sftpSession, path, (FileMode)createMode, FileAccess.Write, (int)_bufferSize, cancellationToken).ConfigureAwait(false))
```

Comment:
Note that to resume an upload using Append mode, the caller will also need to position the input stream at the correct offset, or else the entire input stream will be appended to the remote file.
I added a similar feature to SftpClient in PR #864.

Reply:
This feature is intended to support scenario where one needs to concatenate multiple local files to a single remote file.
Resumption could be done with the new interface, but it is quite hard to do correctly in a general sense. As you already mentioned, the caller would need to seek to correct offset, but the caller first needs to determine what this offset is. So yes, it is now supported by the application needs to handle it correctly.

Reply:
Right. I have been #864's resume feature for a couple of months, it works nicely for both upload and download. But the app does need to check the remote length first to set the proper stream offset.